test/typing/component-function-export.spec.ts

```diff
@@ -75,4 +73,4 @@ export default {
   },
 
   name: 'conditional-slot-parent',
-} as RiotComponentWrapper<ConditionalSlotComponent>
+} as RiotComponentWrapper<ReturnType<typeof exports>>
```

Comment:
This change is not compatible with the current compiler behavior.
Check for example the output of this component here https://riot.js.org/online-compiler/
```riot 
<my-tag>
  <p>Hello</p>
  
  <script>
    export type TestComponent = RiotComponent<{
      name: string
    }, {
      prop: number
    }>
          
    export default {}
  </script>
</my-tag>
```
The compiler should be updated to output this code https://github.com/riot/compiler/blob/main/src/generators/javascript/utils.js#L102